XSLT/chattpubdctermstomods.xsl

```diff
@@ -0,0 +1,148 @@
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-5.xsd">
+                
+                <!-- title -->
+                <xsl:apply-templates select="dcterms:title"/>
+                
+                <!-- identifier -->
+                <location><xsl:apply-templates select="dcterms:identifier"/></location>
+                
+                <!-- description -->
+                <xsl:apply-templates select="dcterms:description"/>
+                
+                <!-- dateCreated -->
+                <xsl:apply-templates select="dcterms:created"/>
+                
+                <!-- subject -->
+                <subject><xsl:apply-templates select="dcterms:subject"/></subject>
+                
+                <!-- recordContentSource -->
+                <recordInfo>
+                    <recordContentSource>Chattanooga Public Library</recordContentSource>
+                </recordInfo>
+                
+                <!-- rights -->
+                <xsl:apply-templates select="dcterms:rights"/>
+                
+            </mods>
+        
+    </xsl:template>
+    
+    
+    <!-- title -->
+    <xsl:template match="dcterms:title">
+        <titleInfo><title><xsl:apply-templates/></title></titleInfo>
+    </xsl:template>
+      
+    <!-- identifiers -->
+    <xsl:template match='dcterms:identifier'>
+        <xsl:choose>
+            <xsl:when test="contains(., 'localhistory')">
+                <url usage="primary" access="object in context"><xsl:apply-templates/></url>
+            </xsl:when>
+        </xsl:choose>
+        <xsl:choose>
+            <xsl:when test="ends-with(., '.pdf')">
+                <url access="preview">https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/PDF_icon.svg/1024px-PDF_icon.svg.png</url>
+            </xsl:when>
+            <xsl:when test="ends-with(., '.jpg')">
+                <xsl:variable name="preview-url" select="replace(., 'original', 'medium')"/>
+                <url access="preview"><xsl:value-of select="$preview-url"/></url>
+            </xsl:when>
+        </xsl:choose>
+    </xsl:template>
+    
+    <!-- description -->
+    <xsl:template match="dcterms:description">
+        <abstract><xsl:apply-templates/></abstract>
+    </xsl:template>
+    
+    <!-- dateCreated -->
+    <xsl:template match="dcterms:created">
+        <xsl:choose>
+            <xsl:when test="contains(., 'circa')"/>
+            <xsl:otherwise>   
+                <originInfo><dateCreated><xsl:apply-templates/></dateCreated></originInfo>
+            </xsl:otherwise> 
+        </xsl:choose>
+    </xsl:template>
+    
+    <!-- subject -->
+    <xsl:template match="dcterms:subject">
+        <xsl:choose>
+            <xsl:when test="ends-with(., '.')">
+                <topic>
+                    <xsl:apply-templates select="substring(., 1, string-length(.) -1)"/>
```

Comment:
```suggestion
                    <xsl:value-of select="substring(., 1, string-length(.) -1)"/>
```

What you've done doesn't work in Saxon8.7.

Reply:
Yep, this change still preserves the change desired (stripping ending periods, especially on single word topics so they have some chance of faceting in DPLA).